src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1222,13 +1221,6 @@ private function registerRouterConfiguration(array $config, ContainerBuilder $co
         if (!class_exists(Psr4DirectoryLoader::class)) {
             $container->removeDefinition('routing.loader.psr4');
         }
-
-        if ($this->isInitializedConfigEnabled('annotations') && (new \ReflectionClass(AnnotationClassLoader::class))->hasProperty('reader')) {
-            $container->getDefinition('routing.loader.annotation')->setArguments([
-                new Reference('annotation_reader'),
-                '%kernel.environment%',
-            ]);
-        }
```

Comment:
Are you sure we can just delete this?

Reply:
I think it's a little too much to think about again. I've updated to use the `routing.loader.attribute` service and keeping the logic.